metis-performance-metric/src/main/java/eu/europeana/metis/performance/metric/model/MetricNumberOfRecordPerOperations.java

```diff
@@ -126,6 +126,13 @@ private int calculateNumberOfRecordsForPluginType(PluginType pluginType, List<Wo
     }
 
     private int getNumberOfRecordInDate(AbstractExecutablePlugin<?> plugin, Date startDate, Date endDate) {
+        if(plugin.getStartedDate() == null){
+            return 0;
+        } else if (plugin.getStartedDate() != null && plugin.getFinishedDate() == null
+                && plugin.getStartedDate().getTime() >=  startDate.getTime()){
+            return plugin.getExecutionProgress().getProcessedRecords() + plugin.getExecutionProgress().getErrors();
+        }
```

Comment:
perphaps writing a comment or a function name for this condition to indicate what means the calculation. Thinking in the future this will be hard to understand or remind ourselves what this condition meant. the following an idea.
```
boolean hasPluginStartTimeDateButNoFinishDateCaseToSumProcessedRecordsAndErrors(AbstractExecutablePlugin<?> plugin) {
return plugin.getStartedDate() != null && plugin.getFinishedDate() == null
                && plugin.getStartedDate().getTime() >=  startDate.getTime();
}
```

Reply:
document all this conditions what they mean. line 139, 144, 150.

Reply:
Created new private method and added comments explaining what is happening

Reply:
excellent